db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2215 @@
+    try(final var statement = connection.createStatement()) {
+      // create the event
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.external_event
+          VALUES ('%s', '%s', '%s', '%s', '%s', '%s');
+          """.formatted(
+              externalEvent.key,
+              externalEvent.event_type_name,
+              externalEvent.source_key,
+              externalEvent.derivation_group_name,
+              externalEvent.start_time,
+              externalEvent.duration
+          )
+      );
+    }
+  }
+
+  protected void associateDerivationGroupWithPlan(int planId, String derivationGroupName) throws SQLException {
+    try(final var statement = connection.createStatement()) {
+      // create the event type
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.plan_derivation_group
+          VALUES ('%s', '%s');
+          """.formatted(planId, derivationGroupName)
+      );
+    }
+  }
+
+  /**
+   * Quick external event creator, leveraging constants from a provided source object (source).
+   */
+  protected ExternalEvent createEvent(String key, String start_time, String duration, ExternalSource source) {
+    return new ExternalEvent(
+        key,
+        et,
+        source.key(),
+        source.derivation_group_name(),
+        start_time,
+        duration
+    );
+  }
+
+  /**
+   * Get all derived events.
+   */
+  protected List<DerivedEvent> getDerivedEvents(String end) throws SQLException {
+    List<DerivedEvent> results = new ArrayList<>();
+    try(final var statement = connection.createStatement()) {
+      var res = statement.executeQuery(
+          // language=sql
+          """
+          SELECT * FROM merlin.derived_events %s
+          """.formatted(end)
+      );
+      while (res.next()) {
+        results.add(new DerivedEvent(
+            res.getString("event_key"),
+            res.getString("event_type_name"),
+            res.getString("source_key"),
+            res.getString("derivation_group_name"),
+            res.getString("start_time"),
+            res.getString("duration"),
+            res.getString("source_range"),
+            res.getString("valid_at")
+        ));
+      }
+    }
+    return results;
+  }
+
+  /**
+   * Repeated function that uploads a source and various events, as well as related types.
+   * Used in:
+   *  - ExternalEventTests.java:
+   *    + BASIC TESTS (derivationGroupComp)
+   *    + final duplication tests (duplicateSource, duplicatedDG)
+   *    + superDerivedEvents
+   *
+   * Data here is based on the SSMO-MPS/mission-data-sandbox/derivation_test examples.
+   */
+  protected void upload_source(String dg) throws SQLException {
+    // First, define the sources.
+    ExternalSource sourceOne = new ExternalSource(
+        "Derivation_Test_00.json",
+        "Test",
```

Comment:
This appears to be replaceable with `st`

Reply:
Generally fixed/replaced in [replace instances of "Test" with st, instances of "Test Default" with dg](https://github.com/NASA-AMMOS/aerie/commit/7058630213577d2ef22ea6e19c01f60a5c6c60a3).